### PR TITLE
Support creating ItemCollections in Transaction Extension

### PR DESCRIPTION
**Pull request by mmcfarland**
**Description:**

Implements "create ItemCollection" via the Transaction extension and provides additional validation and behavior on ids when creating Items or ItemCollections. I'll note that the api spec indicates percent encoded characters are _discouraged_, but this change makes them invalid for this backend. That's an opinioned change, and I'm willing to walk it back to a subset (like just "/") if that flexibility is desired.

See: https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#item-ids
See: https://github.com/radiantearth/stac-api-spec/blob/master/ogcapi-features/extensions/transaction/README.md#methods

Fixes: https://github.com/stac-utils/stac-fastapi-pgstac/issues/23

**PR Checklist:**

- [x] `pre-commit` hooks pass locally
- [x] Tests pass (run `make test`)
- [x] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by mmcfarland** [comment]
_Note_: This is draft until https://github.com/stac-utils/stac-fastapi/pull/574 is merged, as tests here will fail without it.

**Review by gadomski — CHANGES_REQUESTED** [review]

File: stac_fastapi/pgstac/transactions.py [hunk 1]
```diff
@@ -16,33 +17,77 @@
...
+                detail=f"Item ID from path parameter ({expected_item_id}) does not match Item ID from Item ({body_item_id})",
+            )
+
+    async def create_item(
+        self,
+        collection_id: str,
+        item: Union[stac_types.Item, stac_types.ItemCollection],
+        request: Request,
+        **kwargs,
+    ) -> Optional[Union[stac_types.Item, Response]]:
```
[the remark this answers is not in the document]
Comment (mmcfarland) [reply stac_fastapi/pgstac/transactions.py:66]: Based on the [language in the spec](https://github.com/radiantearth/stac-api-spec/blob/master/ogcapi-features/extensions/transaction/README.md#post), it seems the best alignment would be to always return None, then. But it does appear to allow for Item or None depending on the request body type.

When `Item` (emphasis mine)
> _May_ return the content of the newly added resource for a successful operation.

When `ItemCollection`
_unspecified_

The [SQLAlchemy implementation](https://github.com/stac-utils/stac-fastapi-sqlalchemy/blob/main/stac_fastapi/sqlalchemy/transactions.py#L51) does the Item/None response as well.

I think your discomfort + the spec also saying an Item should use a location header pushes me towards always having an empty body, but adding that location header. That a bit of a breaking change (even in many unit tests!) -- are you comfortable with that?

[the remark this answers is not in the document]
Comment (gadomski) [reply stac_fastapi/pgstac/transactions.py:66]: Hm, good point. It's probably not worth breaking just to make things feel nicer ... so let's keep it as is. Thanks for looking into that.

**Review by gadomski — APPROVED** [review]

**Comment by mmcfarland** [comment]
@gadomski I added some commits that should address your comments, thanks for the feedback. I had started down the path of response unification, but then realized the extent to which it was a meaningful breaking changed, and walked it back in a subsequent commit. Tests pass locally against stac-fastapi `main`, so I'll leave this in draft until the next stac-fastapi release.

**Review by gadomski — APPROVED** [review]

**Pull request merged**